### PR TITLE
Fix types for pread/pwrite.

### LINKED ISSUE #131: ARM fix
I recently compiled nix for my beagleboard black - it went smoothly, but I got two compile errors, both in uio.rs.

The parameter `offset` to ffi::pread and ffi::pwrite are i32 on my arm. Maybe they should be isize? I am not sure how to express that in rust in a portable manner.

### PR DESCRIPTION
**Pull request by SirVer**
This fixes compile for 32 bit systems like the ARM I am working on.

Fixes #131.

[Review on Reviewable](https://reviewable.io/reviews/carllerche/nix-rust/134)

### CODE REVIEW

File: src/sys/uio.rs [hunk 1]
```diff
@@ -55,7 +55,7 @@ pub fn readv(fd: RawFd, iov: &mut [IoVec<&mut [u8]>]) -> Result<usize> {
 pub fn pwrite(fd: RawFd, buf: &[u8], offset: i64) -> Result<usize> {
     let res = unsafe {
         ffi::pwrite(fd, buf.as_ptr() as *const c_void, buf.len() as size_t,
-                    offset)
+                    offset as off_t)
```
Comment (posborne) [inline src/sys/uio.rs]: Would it not be better to change the type signature for `pread`/`pwrite` to take offset as an `off_t` rather than casting here?

Comment (carllerche) [inline src/sys/uio.rs]: :+1: I agree. @SirVer Do you want to update it?

**Comment by SirVer** [comment]
Updated.

**Comment by carllerche** [comment]
Thanks! Merged: d9808975608e7818be2d564357469dab0a31e1d2

**Pull request closed without merging**